api/v1beta1/database_webhook.go

```diff
@@ -54,15 +58,40 @@ var _ webhook.Validator = &Database{}
 func (r *Database) ValidateCreate() error {
 	databaselog.Info("validate create", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object creation.
+	err := validateSecretTemplates(r.Spec.SecretsTemplates); if err != nil {
+		return err
+	}
+	
 	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Database) ValidateUpdate(old runtime.Object) error {
 	databaselog.Info("validate update", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object update.
+	err := validateSecretTemplates(r.Spec.SecretsTemplates); if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateSecretTemplates(templates map[string]string) error {
+	for _, template := range templates {
+		allowedFields := []string{"Protocol", "DatabaseHost", "DatabasePort", "UserName", "Password", "DatabaseName"}
+		// This regexp is getting fields from mustache templates so then they can be compared to allowed fields
+		reg := "{{\\s*([\\w\\.]+)\\s*}}"
```

Comment:
why the \s and \w and \. are escaped? Checking with regex101 ... I had first to fix remove them again.